### DISCUSSION TITLE
Must you set up an email service before you can create users in Staging or Prod?

### DISCUSSION THREAD

**Discussion by keepforever** [epicweb-dev/epic-stack#398] [Q&A]
Apologies if I missed it, but, I just want to confirm that, in `staging` and `prod`, new users can NOT sign up UNTIL you hook up the email-sending service, right?

It would be cool to have an escape hatch or "opt-out" for this.  Or perhaps you can, if you want to make sure you don't get spammed with new users by a bot, you have a "secret" code that you only share with the people you want to let sign up before you get around to hooking up Resend.

**Comment by vinstah** [comment]
I would probably use https://mailtrap.io/ for staging, and implement honeypot for anti-bot

**Comment by kentcdodds** [comment]
Hi @keepforever,

Emails will not be sent until you have set up the email provider. If you're ok allowing users to sign up for accounts without first confirming their email address, then you can either remove the verification process or manually add users to the database by [connecting to your prod database](https://github.com/epicweb-dev/epic-stack/blob/main/docs/deployment.md#optional-connecting-to-your-production-database).

If you're only going to have a few users of your app, what I would suggest is removing all the stuff related to sending emails, then, manually create an admin user and adding a page for user management which admin users have access to. Then you can more easily create/update/delete users of the app and even handle password reset etc.

I hope that helps!
